storage/tsdb/tsdb.go

```diff
@@ -236,6 +251,8 @@ func (a appender) Add(lset labels.Labels, t int64, v float64) (uint64, error) {
 	case tsdb.ErrOutOfBounds:
 		return 0, storage.ErrOutOfBounds
 	}
+	m := *(a.lastSamplesAddTime)
+	m.SetToCurrentTime()
```

Comment:
This is pretty expensive to do on every single ingested sample.

Reply:
Agreed. If you have suggestions for another way to get the same or similar data I'm open to it. The use case is in https://github.com/prometheus/tsdb/pull/401, but the short of it is that I would like to be able to know how far behind the local storage (in terms of storing metrics) a given remote write is by comparing the last time we ingested samples vs the last time we successfully sent samples.

Reply:
Queue size would be the usual way to do that.

Reply:
The problem with that is that the queue mechanics are changing as part of my remote write via WAL PR.